torch/optim/adam.py

```diff
@@ -315,14 +342,15 @@ def _single_tensor_adam(params: List[Tensor],
                         amsgrad: bool,
                         beta1: float,
                         beta2: float,
-                        lr: float,
+                        lr: Union[float, Tensor],
                         weight_decay: float,
                         eps: float,
                         maximize: bool,
                         capturable: bool,
                         differentiable: bool):
 
     assert grad_scale is None and found_inf is None
+    assert not isinstance(lr, Tensor), "lr should have been coerced to a scalar in _single_tensor_adam"
```

Comment:
nit: in -> before calling

Reply:
Sorry I don't know how to interpret your recommendation 😬